syntax = "proto3";
package demo;

service UserService {
  rpc GetAllUsers (Empty) returns (UserList) {}
  rpc GetAllSongs (Empty) returns (SongList) {}
  rpc GetUserPlaylists (IdRequest) returns (PlaylistList) {}
  rpc GetPlaylistSongs (IdRequest) returns (SongList) {}
  rpc GetPlaylistsBySong (IdRequest) returns (PlaylistList) {}
}

message Empty {}
message IdRequest { int32 id = 1; }

message Song {
    int32 id = 1;
    string title = 2;
    string artist = 3;
}

message Playlist {
    int32 id = 1;
    string name = 2;
    repeated Song songs = 3;
}

message UserResponse {
    int32 id = 1;
    string name = 2;
    int32 age = 3;
    repeated Playlist playlists = 4;
}

message SongList { repeated Song songs = 1; }
message PlaylistList { repeated Playlist playlists = 1; }
message UserList { repeated UserResponse users = 1; }
